lib/puppet/module_tool/applications/builder.rb

```diff
@@ -135,9 +135,9 @@ def write_json
           f.write(metadata.to_json)
         end
 
-        # PSON.pretty_generate is a BINARY string
+        # JSON.pretty_generate is a BINARY string
         Puppet::FileSystem.open(File.join(build_path, 'checksums.json'), nil, 'wb') do |f|
-          f.write(PSON.pretty_generate(Checksums.new(build_path)))
+          f.write(JSON.pretty_generate(Checksums.new(build_path)))
```

Comment:
I think the same comment applies - the Checksums class mixes in `FormatSupport` and has a `:to_data_hash` method so the `to_json` method on `FormatSupport` actually lets this work:

```ruby
12] pry(main)> checksums = Puppet::ModuleTool::Checksums.new('/Users/moses/foo/bar')
=> #<Puppet::ModuleTool::Checksums:0x007facd0b90170 @path=#<Pathname:/Users/moses/foo/bar>>
[13] pry(main)> JSON::Parser
=> JSON::Ext::Parser
[14] pry(main)> JSON.pretty_generate(checksums)
=> "{\n  \"baz\": \"68b329da9893e34099c7d8ad5cb9c940\"\n}"
[15] pry(main)> checksums.to_json
=> "{\"baz\":\"68b329da9893e34099c7d8ad5cb9c940\"}"
```

Without `to_json` defined on FormatSupport, we get this in either case:
```ruby
[5] pry(main)> JSON.pretty_generate(checksums)
JSON::GeneratorError: only generation of JSON objects or arrays allowed
from /Users/moses/.rbenv/versions/2.3.1/lib/ruby/2.3.0/json/common.rb:286:in `generate'
```

(json_pure has been removed)